src/com/seafile/seadroid2/ui/activity/OpenFileBaseActivity.java

```diff
@@ -0,0 +1,98 @@
+package com.seafile.seadroid2.ui.activity;
+
+import android.content.*;
+import android.net.Uri;
+import android.os.IBinder;
+import android.webkit.MimeTypeMap;
+import com.actionbarsherlock.app.SherlockFragmentActivity;
+import com.seafile.seadroid2.R;
+import com.seafile.seadroid2.account.Account;
+import com.seafile.seadroid2.data.DataManager;
+import com.seafile.seadroid2.transfer.TransferService;
+import com.seafile.seadroid2.util.ToastUtils;
+import com.seafile.seadroid2.ui.dialog.OpenAsDialog;
+
+import java.io.File;
+
+/**
+ * Base class for opening a file
+ */
+public abstract class OpenFileBaseActivity extends SherlockFragmentActivity {
+
```

Comment:
Emm.. I don't think it's a good idea to have this base activity class and let `FileActivity` and `SearchActivity` to inherit from it.

I think refactor out the common functions (`showFile`, `startMarkdownActivity`, etc.) to a util class is enough.